tiny_example.snakefile

```diff
@@ -0,0 +1,320 @@
+        # gather RNAseq sample
+        #expand("outputs/rnaseq_sourmash_gather/{sra}_gtdb_k31.csv", sra = SRA)
+        Checkpoint_RnaseqToReference(expand("outputs/rnaseq_salmon/{{gtdb_species}}-{sra}_quant/quant.sf", sra = SRA))
+
+##############################################################
+## Generate reference transcriptome using pangenome analysis
+##############################################################
+
+rule download_gtdb_lineages:
+    output: "inputs/gtdb-rs202.taxonomy.v2.csv"
+    resources:
+        mem_mb = 4000
+    threads: 1
+    shell:"""
+    wget -O {output} https://osf.io/p6z3w/download
+    """
+ 
+# The outputs of this rule are used by the class Checkpoint_GrabAccessions.
+# The new wildcard, acc, is encoded in the "ident" column of the output
+# file "accessions". 
+checkpoint grab_species_accessions:
+    input: gtdb_lineages = "inputs/gtdb-rs202.taxonomy.v2.csv"
+    output: 
+        accessions = "outputs/gtdb_genomes_by_species/{gtdb_species}.csv",
+        charcoal_lineages = "outputs/gtdb_genomes_by_species/{gtdb_species}_charcoal_lineages.csv"
+    resources:
+        mem_mb = 4000
+    threads: 1
+    conda: "envs/tidyverse.yml"
+    script: "scripts/grab_species_accessions.R"
+
+rule make_genome_info_csv:
+    output:
+        csv = 'inputs/gtdb_genomes/{gtdb_species}/{acc}.info.csv'
+    conda: "envs/download_genomes.yml"
+    resources:
+        mem_mb = 8000
+    threads: 1
+    shell: """
+    python scripts/genbank_genomes.py {wildcards.acc} --output {output.csv}
+    """
+
+rule download_matching_genomes:
+    input:
+        csvfile = ancient('inputs/gtdb_genomes/{gtdb_species}/{acc}.info.csv')
+    output:
+        genome = "inputs/gtdb_genomes/{gtdb_species}/{acc}_genomic.fna.gz"
+    resources:
+        mem_mb = 500
+    threads: 1
+    run:
+        with open(input.csvfile, 'rt') as infp:
+            r = csv.DictReader(infp)
+            rows = list(r)
+            assert len(rows) == 1
+            row = rows[0]
+            acc = row['acc']
+            assert wildcards.acc.startswith(acc)
+            url = row['genome_url']
+            name = row['ncbi_tax_name']
+
+            print(f"downloading genome for acc {acc}/{name} from NCBI...",
+                file=sys.stderr)
+            with open(output.genome, 'wb') as outfp:
+                with urllib.request.urlopen(url) as response:
+                    content = response.read()
+                    outfp.write(content)
+                    print(f"...wrote {len(content)} bytes to {output.genome}",
+                        file=sys.stderr)
+
+
+rule cat_genome_sequences:
+    input: Checkpoint_GrabAccessions("inputs/gtdb_genomes/{{gtdb_species}}/{acc}_genomic.fna.gz")
+    output: "outputs/gtdb_genomes_cat/{gtdb_species}.fna.gz"
+    threads: 1
+    resources: 
+        mem_mb=2000,
+        tmpdir=TMPDIR
+    benchmark:"benchmarks/cat_annotated_seqs_{gtdb_species}.txt"
+    shell:'''
+    cat {input} > {output}
```

Comment:
Are there potentially multiple accessions per species? Will this overwrite the output file instead of appending to it?

Reply:
There are almost always multiple accessions per species -- this would concatenate all inputs into a single output file. The syntax is a little weird, where if there are multiple inputs, snakemake sees them as a list, but will give them all to bash separated by a space. In any case, as long as this produces _a_ file, for the purposes of this test file it's fine. The error I was getting was related to the DAG creation and didn't get to the code running step 😆